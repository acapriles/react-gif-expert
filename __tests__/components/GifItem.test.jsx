import { getByRole, render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem />', () => {
	const properties = {
		id: 1,
		title: 'Dragon Ball',
		url: 'https://media3.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=4ff38d30e3sg5o7ov5jvitl2bmssc3rnbnvig7nbsx8xqgei&rid=giphy.gif&ct=g',
	};

	const { title, url } = properties;

	test('debe hacer match con el snapshot', () => {
		const { container } = render(<GifItem {...properties} />);

		expect(container).toMatchSnapshot();
	});

	test('debe mostrar la imagen con el URL y el ALT indicado', () => {
		const { container } = render(<GifItem {...properties} />);
		// screen.debug();

		// expect(screen.getByRole('img').src).toBe( url );
		// expect(screen.getByRole('img').alt).toBe( title );

		const { src, alt } = screen.getByRole('img');
		expect(src).toBe(url);
		expect(alt).toBe(title);
	});

	test('debe mostrar el tÃ­tulo en el componente', () => {
		const { container } = render(<GifItem {...properties} />);

		expect(screen.getByText(title)).toBeTruthy();
	});
});
