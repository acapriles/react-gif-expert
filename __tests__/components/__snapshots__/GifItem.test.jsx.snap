// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="https://media3.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=4ff38d30e3sg5o7ov5jvitl2bmssc3rnbnvig7nbsx8xqgei&rid=giphy.gif&ct=g"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
